File: rules/java/shor-broken.java
// Test fixtures for pqc-semgrep-rules / Java / Shor-broken cryptography.
import java.security.KeyPairGenerator;
import java.security.MessageDigest;
import java.security.Signature;
import javax.crypto.KeyAgreement;

public class ShorBrokenFixture {
    public static void main(String[] args) throws Exception {
        // ---- RSA ----
        // ruleid: pqc.java.rsa
        KeyPairGenerator rsa1 = KeyPairGenerator.getInstance("RSA");
        // ruleid: pqc.java.rsa
        Signature rsaSig = Signature.getInstance("SHA256withRSA");

        // ---- EC ----
        // ruleid: pqc.java.ec
        KeyPairGenerator ec1 = KeyPairGenerator.getInstance("EC");
        // ruleid: pqc.java.ec
        KeyAgreement ecdh = KeyAgreement.getInstance("ECDH");
        // ruleid: pqc.java.ec
        Signature ecSig = Signature.getInstance("SHA256withECDSA");

        // ---- DSA ----
        // ruleid: pqc.java.dsa
        KeyPairGenerator dsa1 = KeyPairGenerator.getInstance("DSA");
        // ruleid: pqc.java.dsa
        Signature dsaSig = Signature.getInstance("SHA256withDSA");

        // ---- DH ----
        // ruleid: pqc.java.dh
        KeyPairGenerator dh1 = KeyPairGenerator.getInstance("DiffieHellman");
        // ruleid: pqc.java.dh
        KeyAgreement dhKa = KeyAgreement.getInstance("DH");

        // Safe (should NOT match these rules)
        // ok: pqc.java.rsa
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
    }
}

File: rules/javascript/shor-broken.ts
// Test fixtures for pqc-semgrep-rules / TypeScript / Shor-broken cryptography.
// Same patterns as shor-broken.js but with TypeScript-specific syntax
// (explicit types, `as` casts) to verify the rules match through the TS
// parser path as well.

import * as crypto from "crypto";

function vulnerable(): void {
    // ---- RSA ----
    // ruleid: pqc.javascript.rsa
    const rsa1 = crypto.generateKeyPairSync("rsa", { modulusLength: 2048 });
    // ruleid: pqc.javascript.rsa
    crypto.generateKeyPair(
        "rsa",
        { modulusLength: 2048 },
        (err: Error | null, publicKey: crypto.KeyObject, privateKey: crypto.KeyObject) => {},
    );

    // ---- EC ----
    // ruleid: pqc.javascript.ec
    const ec1 = crypto.generateKeyPairSync("ec", { namedCurve: "P-256" });
    // ruleid: pqc.javascript.ec
    const ecdh: crypto.ECDH = crypto.createECDH("secp256k1");

    // ---- DSA ----
    // ruleid: pqc.javascript.dsa
    const dsa1 = crypto.generateKeyPairSync("dsa", {
        modulusLength: 2048,
        divisorLength: 256,
    });

    // ---- DH ----
    // ruleid: pqc.javascript.dh
    const dh1: crypto.DiffieHellman = crypto.createDiffieHellman(2048);
    // ruleid: pqc.javascript.dh
    const dh2 = crypto.getDiffieHellman("modp14");

    // ---- X25519 ----
    // ruleid: pqc.javascript.x25519
    const x = crypto.generateKeyPairSync("x25519");

    // ---- Ed25519 ----
    // ruleid: pqc.javascript.ed25519
    const ed = crypto.generateKeyPairSync("ed25519");
}

function safe(): void {
    // ok: pqc.javascript.rsa
    const buf: Buffer = crypto.randomBytes(32);
}
